api/spec/requests/spree/api/v1/variant_cache_spec.rb

```diff
@@ -0,0 +1,120 @@
+      Spree::Variant.cache_versioning = true
+
+      expect(Spree::Variant.cache_versioning).to be_truthy
+      expect(Spree::Variant.collection_cache_versioning).to be_truthy
+    end
+
+    context 'without stock' do
+      before { subject.stock_items.first.set_count_on_hand(0) }
+
+      it 'returns an out of stock response' do
+        get "/api/v1/variants/#{subject.id}", params: { token: user.spree_api_key }
+        expect(response.status).to eq(200)
+
+        variant = JSON.parse(response.body)
+
+        expect(variant['stock_items'].size).to eq(1)
+        expect(variant['total_on_hand']).to be(0)
+        expect(variant['is_orderable']).to be_falsey
+        expect(variant['is_backorderable']).to be_falsey
+        expect(variant['in_stock']).to be_falsey
+      end
+    end
+
+    context 'with stock' do
+      before { subject.stock_items.first.set_count_on_hand(5) }
+
+      it 'returns a positive in stock response' do
+        get "/api/v1/variants/#{subject.id}", params: { token: user.spree_api_key }
+        expect(response.status).to eq(200)
+
+        variant = JSON.parse(response.body)
+
+        expect(variant['stock_items'].size).to eq(1)
+        expect(variant['total_on_hand']).to be(5)
+        expect(variant['is_orderable']).to be_truthy
+        expect(variant['is_backorderable']).to be_falsey
+        expect(variant['in_stock']).to be_truthy
+      end
+    end
+  end
+
+
+  context 'with cache versioning turned off' do
+    before do
+      Spree::Variant.collection_cache_versioning = false
+      Spree::Variant.cache_versioning = false
+
+      expect(Spree::Variant.cache_versioning).to be_falsey
+      expect(Spree::Variant.collection_cache_versioning).to be_falsey
+    end
+
+    context 'without stock' do
+      before { subject.stock_items.first.set_count_on_hand(0) }
+
+      it 'returns an out of stock response' do
+        get "/api/v1/variants/#{subject.id}", params: { token: user.spree_api_key }
+        expect(response.status).to eq(200)
+
+        variant = JSON.parse(response.body)
+
+        expect(variant['stock_items'].size).to eq(1)
+        expect(variant['total_on_hand']).to be(0)
+        expect(variant['is_orderable']).to be_falsey
+        expect(variant['is_backorderable']).to be_falsey
+        expect(variant['in_stock']).to be_falsey
+      end
+    end
+
+    context 'with stock' do
+      before { subject.stock_items.first.set_count_on_hand(5) }
+
+      it 'returns a positive in stock response' do
+        get "/api/v1/variants/#{subject.id}", params: { token: user.spree_api_key }
+        expect(response.status).to eq(200)
+
+        variant = JSON.parse(response.body)
+
+        expect(variant['stock_items'].size).to eq(1)
+        expect(variant['total_on_hand']).to be(5)
+        expect(variant['is_orderable']).to be_truthy
+        expect(variant['is_backorderable']).to be_falsey
+        expect(variant['in_stock']).to be_truthy
+      end
+    end
+  end
+
+
+
```

Comment:
```suggestion
```

Reply:
I  have removed the empty space. Wasn't that clear in the IDE because of block separation lines which get squashed by git.